### PR TITLE
Return an http error when the user try to set an unsupported temporarily configuation

### PR DESCRIPTION
**Pull request by longfangsong**
Signed-off-by: longfangsong <longfangsong@icloud.com>

### What problem does this PR solve?

Part of #3220

### What is changed and how it works?

Check whether a temporarily configuation is supported. If not, return an error on `SaveTTLConfig` and return a `StatusBadRequest` to the user.

### Check List

Tests

- Unit test

Code changes

- Has configuration change
- Has HTTP API interfaces change

### Release note

- No release note

### CODE REVIEW

**Comment by longfangsong** [comment]
@Yisaer PTAL

File: server/server.go [hunk 1]
```diff
@@ -1292,6 +1292,11 @@ func (s *Server) PersistFile(name string, data []byte) error {
 
 // SaveTTLConfig save ttl config
 func (s *Server) SaveTTLConfig(data map[string]interface{}, ttl time.Duration) error {
+	for k := range data {
+		if !config.IsSupportedTTLConfig(k) {
+			return errors.New("unsupported ttl config " + k)
```
Comment (Yisaer) [inline server/server.go:1297]: use `fmt.Errof` instead.

File: server/api/config_test.go [hunk 2]
```diff
@@ -313,4 +317,9 @@ func (s *testConfigSuite) TestConfigTTL(c *C) {
 	assertTTLConfig(c, s.svr.GetPersistOptions(), Equals)
 	time.Sleep(2 * time.Second)
 	assertTTLConfig(c, s.svr.GetPersistOptions(), Not(Equals))
+
+	postData, err = json.Marshal(invalidTTLConfig)
+	c.Assert(err, IsNil)
+	err = postJSON(testDialClient, addr, postData)
+	c.Assert(err, Not(IsNil))
```
Comment (Yisaer) [inline server/api/config_test.go:324]: Also assert the error message here.

**Review by Yisaer — COMMENTED** [review]
plz address the comment

**Review by Yisaer — APPROVED** [review]
LGTM

**Review by nolouch — APPROVED** [review]
lgtm

**Review by HunDunDM — APPROVED** [review]

**Pull request merged**